tests/test_mpls.py

```diff
@@ -0,0 +1,1049 @@
+        # set ip address
+        self.add_ip_address("Ethernet0", "10.0.0.0/31")
+        self.add_ip_address("Ethernet4", "10.0.0.2/31")
+        self.add_ip_address("Ethernet4", "10.0.0.4/31")
+
+        # bring up interface
+        self.set_admin_status("Ethernet0", "up")
+        self.set_admin_status("Ethernet4", "up")
+        self.set_admin_status("Ethernet8", "up")
+
+        # set ip address and default route
+        dvs.servers[0].runcmd("ip address add 10.0.0.1/31 dev eth0")
+        dvs.servers[0].runcmd("ip route add default via 10.0.0.0")
+
+        dvs.servers[1].runcmd("ip address add 10.0.0.3/31 dev eth0")
+        dvs.servers[1].runcmd("ip route add default via 10.0.0.2")
+
+        dvs.servers[2].runcmd("ip address add 10.0.0.5/31 dev eth0")
+        dvs.servers[2].runcmd("ip route add default via 10.0.0.4")
+
+        # get neighbor and arp entry
+        dvs.servers[0].runcmd("ping -c 1 10.0.0.3")
+        dvs.servers[2].runcmd("ping -c 1 10.0.0.3")
+
+        # add route entry
+        # dvs.runcmd("ip -f mpls route add 200 nexthop as 201 via inet 10.0.0.1 dev Ethernet0 nexthop as 202 via inet 10.0.0.5 dev Ethernet8")
+        label = "200"
+        fieldValues = {"nexthop": "10.0.0.1,10.0.0.5", "ifname": "Ethernet0,Ethernet8", "mpls_nh" : "swap+201,swap+202", "mpls_pop" : "1"}
+        self.create_inseg_entry(label, fieldValues)
+
+        # check application database
+        self.pdb.wait_for_entry("LABEL_ROUTE_TABLE", label)
+
+        # check ASIC inseg database
+        self.check_inseg_entries([label])
+
+        # remove route entry
+        # dvs.runcmd("ip -f mpls route del 200 nexthop as 201 via inet 10.0.0.1 dev Ethernet0 nexthop as 202 via inet 10.0.0.5 dev Ethernet8")
+        self.remove_inseg_entry(label)
+
+        # check application database
+        self.pdb.wait_for_deleted_entry("LABEL_ROUTE_TABLE", label)
+
+        # check ASIC inseg database
+        self.check_deleted_inseg_entries([label])
+
+        # remove ip address
+        self.remove_ip_address("Ethernet0", "10.0.0.0/31")
+        self.remove_ip_address("Ethernet4", "10.0.0.2/31")
+        self.remove_ip_address("Ethernet8", "10.0.0.4/31")
+
+        # remove mpls interface
+        self.remove_mpls_intf("Ethernet0")
+        self.remove_mpls_intf("Ethernet4")
+        self.remove_mpls_intf("Ethernet8")
+
+        self.set_admin_status("Ethernet0", "down")
+        self.set_admin_status("Ethernet4", "down")
+        self.set_admin_status("Ethernet8", "down")
+
+        # remove ip address and default route
+        dvs.servers[0].runcmd("ip route del default dev eth0")
+        dvs.servers[0].runcmd("ip address del 10.0.0.1/31 dev eth0")
+
+        dvs.servers[1].runcmd("ip route del default dev eth0")
+        dvs.servers[1].runcmd("ip address del 10.0.0.3/31 dev eth0")
+
+        dvs.servers[2].runcmd("ip route del default dev eth0")
+        dvs.servers[2].runcmd("ip address del 10.0.0.5/31 dev eth0")
+
+        self.clear_mpls(dvs)
+
```

Comment:
Can you add a test case for 'unresolved ARP' entry. And ensure Nexthop is not added.  And then resolve the ARP entry and ensure nexthop is added.